flink-table-store-core/src/test/java/org/apache/flink/table/store/file/TestFileStore.java

```diff
@@ -314,6 +315,12 @@ public void assertCleaned() {
         } catch (IOException e) {
             throw new RuntimeException(e);
         }
+
+        // remove best effort latest and earliest hint files
+        Path snapshotDir = pathFactory().snapshotDirectory();
+        actualFiles.remove(new Path(snapshotDir, SnapshotFinder.LATEST));
+        actualFiles.remove(new Path(snapshotDir, SnapshotFinder.EARLIEST));
```

Comment:
Check their values are valid instead of removing them.

Reply:
If a concurrency test is added in the future, it will not be possible to check here because the hint_file is possibly not the most accurate, so this check is only.
latest should < true_latest
earliest should < true_earliest

Reply:
No, latest must be equal to the ground truth. It is only changed by the commit operation and that value should be precise.

Reply:
It looks like we need a precise value from `findLatest` and `findEarliest`.
We can do it.